### PR TITLE
Features/ROC 8274 additional DAC accessibility test scripts

### PR DESCRIPTION
**Pull request by pradhap87**
### JIRA link

https://tools.hmcts.net/jira/browse/ROC-8274
https://tools.hmcts.net/jira/browse/ROC-8306

### Change description

Additional DAC accessibility test scripts as part of DAC testing. This PR will covers all the sub-tasks under [ROC-8274](https://tools.hmcts.net/jira/browse/ROC-8274)

### Work checklist

- [ ] Unit tests added where applicable
- [ ] Route tests added for new pages
- [ ] New pages included in a11y tests
- [ ] Task list and task completeness checks updated
- [ ] Check and send page updated
- [ ] Routes, page content and page flows of new features are toggled 
- [ ] UI changes look good on mobile
- [ ] Required Google Analytics events are being sent 

### Developer self-QA run statement

- [x] I have clicked through the running application to see if all changes I made actually work.

### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/test/a11y/customChecks.ts [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+}
+
+export const checkRole = (window: Window, document: Document) => {
+  // this expects and validates 'Total monthly income' 'p' tag has role="status" and aria-live="polite"
+  /**
+   * <p class="heading-small calculation-outcome-container" role="status" aria-live="polite">
+   *  {{ t('Total monthly income:') }} £<span class="total-monthly-income-expense"> {{ totalMonthlyIncomeExpense | default('0.00') }} </span>
+   * </p>
+   */
+  const definitionList = document.getElementsByClassName('calculation-outcome-container')
```
Comment (prudhvi-v) [inline src/test/a11y/customChecks.ts:125]: Can we change const name from 'definitionList' to something more appropriate to the test like 'messageElement' or 'elementWithTotal'

File: src/test/a11y/customChecks.ts [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+  expect(definitionList.length, 'Income / Expence page must have role="status" and aria-live="polite" for the "p" tag').to.be.greaterThan(0)
+
+  for (let i = 0; i < definitionList.length; i++) {
+    const attributesName = definitionList[i].getAttributeNames()
+    expect(attributesName.length,'Total Income / Expence "p" tag must have "role" & "aria-live" attributes').to.be.greaterThan(2)
+    expect(attributesName).to.contains('role')
+    expect(attributesName).to.contains('aria-label')
```
Comment (prudhvi-v) [inline src/test/a11y/customChecks.ts]: Does the 'p' tag has 'aria-label' ?. As per the comments there is no 'aria-label' for <p... tag>

File: src/test/a11y/customChecks.ts [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+    }
+  }
+}
+
+export const checkButton = (window: Window, document: Document) => {
+  // this expects and validates 'aria-label' attribute is present in the submit button"
+  /**
+   * <input type="submit" class="button" aria-label="I confirm I’ve read this information about resolving disputes" value="I confirm I’ve read this">
+   */
+  const definitionList = document.getElementsByClassName('button')
```
Comment (prudhvi-v) [inline src/test/a11y/customChecks.ts:145]: please change the variable name to more appropriate one for this test case.

File: src/test/a11y/customChecks.ts [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+  expect(definitionList.length, 'resolving-this-dispute page must have submite button').to.be.greaterThan(0)
```
Comment (prudhvi-v) [inline src/test/a11y/customChecks.ts:146]: I believe this page should have only one button so .. expect(...).to.be.equal(1) would be more appropriate. What do you think?

File: src/test/a11y/customChecks.ts [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+
+  for (let i = 0; i < definitionList.length; i++) {
```
Comment (prudhvi-v) [inline src/test/a11y/customChecks.ts:148]: I don't think we need for loop here. Since we always expect 1 input button

**Review by prudhvi-v — CHANGES_REQUESTED** [review]

File: src/test/a11y/customChecks.ts (hunk 5 above)
Comment (pradhap87) [reply src/test/a11y/customChecks.ts]: removed the loop 
image

File: src/test/a11y/customChecks.ts (hunk 1 above)
Comment (pradhap87) [reply src/test/a11y/customChecks.ts:125]: done updated as **elementWithTotal**

File: src/test/a11y/customChecks.ts (hunk 2 above)
Comment (pradhap87) [reply src/test/a11y/customChecks.ts]: it's aria-live, typo i have updated it

File: src/test/a11y/customChecks.ts (hunk 3 above)
Comment (pradhap87) [reply src/test/a11y/customChecks.ts:145]: updated

File: src/test/a11y/customChecks.ts (hunk 4 above)
Comment (pradhap87) [reply src/test/a11y/customChecks.ts:146]: agreed :) Updated

**Comment by pradhap87** [comment]
This branch has been merged with branch ROC-8267 (PR 2474). Hence closing this PR.

image

**Pull request closed without merging**